framework/src/base/NonlinearSystemBase.C

```diff
@@ -2308,6 +2314,69 @@ NonlinearSystemBase::residualGhosted()
 }
 
 void
+NonlinearSystemBase::updateJacobianSparsity(SparseMatrix<Number> & jacobian)
```

Comment:
This whole routine looks a lot like the `augmentSparsity` routine. Can we factor out common parts into another routine?